### PR TITLE
Rename coords types

### PR DESCRIPTION
**Pull request by ZehMatt**
This makes it a lot easier to type and write. Now its also very distinct between Pos2 and Pos3 where for Pos and Pos3 someone not familiar with the code base can be easily tricked.

### CODE REVIEW

File: src/OpenLoco/Audio/Audio.h [hunk 1]
```diff
@@ -99,10 +99,10 @@ namespace OpenLoco::Audio
     void pauseSound();
     void unpauseSound();
     void playSound(Vehicles::Vehicle2or6* t);
-    void playSound(sound_id id, Map::map_pos3 loc);
-    void playSound(sound_id id, Map::map_pos3 loc, int32_t pan);
+    void playSound(sound_id id, Map::Pos3 loc);
+    void playSound(sound_id id, Map::Pos3 loc, int32_t pan);
```
Comment (duncanspumpkin) [inline src/OpenLoco/Audio/Audio.h:103]: whilst we are at it should we perhaps make these `const &`

File: src/OpenLoco/Entities/Entity.h [hunk 2]
```diff
@@ -76,7 +76,7 @@ namespace OpenLoco
         CompanyId_t owner; // 0x21
         string_id name;    // 0x22, combined with ordinalNumber on vehicles
 
-        void moveTo(Map::map_pos3 loc);
+        void moveTo(Map::Pos3 loc);
```
Comment (duncanspumpkin) [inline src/OpenLoco/Entities/Entity.h:79]: likewise `const &`

**Review by duncanspumpkin — APPROVED** [review]
This all looks sensible. Perhaps we should take the opportunity of trying to enforce `const &` for passing these variables.

**Review by AaronVanGeffen — APPROVED** [review]
Looks like a nice cleanup to me.

**Review by marijnvdwerf — APPROVED** [review]
Looks alright to me. Makes me wonder how well we split world coordinates from UI/render coordinates...

**Comment by ZehMatt** [comment]
> Looks alright to me. Makes me wonder how well we split world coordinates from UI/render coordinates...

Ideally the type comes from the module "UI"

So the structure could be
```
map\
map\coords.hpp
```
```
ui\
ui\coords.hpp
```
etc.

This also avoids having to include the wrong things u dont need.

File: src/OpenLoco/Audio/Audio.h (hunk 1 above)
Comment (ZehMatt) [reply src/OpenLoco/Audio/Audio.h:103]: Yea, I'll do it later today.

**Pull request merged**